### ISSUE TITLE
Unexpected behavior with Google `pydocstyle` convention

### ISSUE THREAD

**Issue by jbschiratti** [astral-sh/ruff#20890]
### Summary

I apologize in advance if this issue is a misunderstanding of the expected behavior of `ruff`.

I created a `dummy.py` Python file with the following content:
```python
"""Dummy Python script."""


def function_with_numpy_style_docstring(value: int, name: str = "John") -> str:
    """Determines how much money a person has.

    Parameters
    ----------
    value : int
        The amount of money a person has.
    name : str
        The name of the person.

    Returns:
    -------
    str:
        A short sentence.
    """
    return f"{name} has {value:d} dollars."
```

The file was created in an empty folder. It's not part of any project (i.e. no `.ruff.toml` or `pyproject.toml` files).

I ran `ruff check` as follows:
```bash
ruff check --select="D" --config 'lint.pydocstyle.convention = "google"' dummy.py
```
which returned:
```bash
All checks passed!
```

To me, the docstring above does not comply with the [Google style](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).

When running `ruff check` on this file, I would have expected an error. Am I missing something obvious here?

### Version

ruff 0.14.0

**Comment by ntBre** [comment]
Ruff would only emit a diagnostic if one of the specific [`D` rules](https://docs.astral.sh/ruff/rules/#pydocstyle-d) is broken, which may not be the case here. We don't have a more general "enforce Google style" rule, so I think this might be the expected behavior. We do flag some issues if the convention is unspecified, though: [playground](https://play.ruff.rs/311681a4-1c3d-44fd-9c96-c7c35b8dcc79). Maybe we fall back on numpy style?

**Comment by MichaReiser** [comment]
> To me, the docstring above does not comply with the [Google style](https://google.github.io/styleguide/pyguide.html#38-comments-and-docstrings).

Can you say more why this docstring isn't following the google style convention as I'm not that familiar with that convention.
